cmake/Options/TileDBAssertions.cmake

```diff
@@ -8,10 +8,10 @@ if(TILEDB_ASSERTIONS)
   endif()
   # On non-Debug builds cmake automatically defines NDEBUG, so we
   # explicitly undefine it:
+  add_compile_options($<$<AND:$<NOT:$<CONFIG:Debug>>,$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>>:-UNDEBUG>)
   if( NOT CMAKE_BUILD_TYPE STREQUAL "Debug" )
```

Comment:
The predicate to look for debug builds doesn't take into account `RelWithDebInfo`. It was added relatively recently, but this statement was missed then.

This is pre-existing defect, but it should be fixed with this PR.

Reply:
Fixed. Also what to do with the `if( NOT CMAKE_BUILD_TYPE STREQUAL "Debug" )` below? I don't think we can make it work in a multi-config generator.